python/pyarrow/table.pxi

```diff
@@ -1450,8 +1450,76 @@ cdef _sanitize_arrays(arrays, names, schema, metadata,
             converted_arrays.append(item)
     return converted_arrays
 
+cdef class _Table(_PandasConvertible):
+    """Internal: An interface for common table operations."""
 
-cdef class RecordBatch(_PandasConvertible):
+    def __init__(self):
+        raise TypeError("This object is not instantiable, "
+                        "use a subclass instead.")
+
+    def drop_null(self):
+        """
+        Remove missing values from a Table.
```

Comment:
```suggestion
        Remove missing values from a RecordBatch or Table.
```

Do we always want to mention both like this?

Reply:
Updated. I could also change to `Remove missing values from tabular object.`